src/data/sparse_page_source.cc

```diff
@@ -6,19 +6,44 @@
 #include <dmlc/timer.h>
 #include <xgboost/logging.h>
 #include <memory>
+#include <vector>
+#include <string>
+#include <locale>
 
 #if DMLC_ENABLE_STD_THREAD
 #include "./sparse_page_source.h"
 #include "../common/common.h"
 
+namespace {
+
+// Split a cache info string with delimiter ':'
+// If cache info string contains drive letter (e.g. C:), exclude it before splitting
+inline std::vector<std::string>
+GetCacheShards(const std::string& cache_info) {
+  if (cache_info.length() < 2) {
+    return {cache_info};
+  }
+  if (std::isalpha(cache_info[0], std::locale::classic())
```

Comment:
I'm not sure, is it a guarantee that path like "A:xxx" and A must be drive letter?

Reply:
Currently I'm quite limited to Linux.  I will need to spend some more time on Windows. :(

Reply:
In Windows, the colon is a reserved letter and thus it cannot appear inside the path. Only time it appears is as part of drive letter. See https://docs.microsoft.com/en-us/windows/desktop/fileio/naming-a-file#naming-conventions for more information about Windows paths.

To be safe, I'll add a guard to this check so that it only runs for Windows.

Reply:
Thanks for the clarification.  With the guard then PR looks good to me. :)